src/Corrade/Utility/Sha1.h

```diff
@@ -36,13 +36,22 @@
 
 namespace Corrade { namespace Utility {
 
-/** @brief SHA-1 */
+/**
+@brief SHA-1
+
+Implementation of the [Secure Hash Algorithm 1](https://en.wikipedia.org/wiki/SHA-1).
+Example usage:
+
+@snippet Utility.cpp Sha1-usage
+
+*/
 class CORRADE_UTILITY_EXPORT Sha1: public AbstractHash<20> {
     public:
         /**
          * @brief Digest of given data
          *
          * Convenience function for @cpp (Utility::Sha1{} << data).digest() @ce.
+         * The @cpp '\0' @ce delimiter is not included.
```

Comment:
I wouldn't mention this here at all, it's misleading -- what it does do when your string has a 0 byte in the middle? Or when `data.back() == '\0'`? The std::string historically didn't have to allocate extra space for the 0-delimiter (it does so only from c++11).

Same below and in the snippet above, it only adds confusion IMHO -- it takes all `size()` bytes from it, which is what one expects, so it doesn't need to be mentioned explicitly.

Reply:
It cause a lot of confusion for me not knowing whether or not it does include the final delimiter and I originally expected it to, so having this somewhere would make sense to me... maybe as a note? If there is `\0` in the middle, it will already mess up there with the char* => string conversion and use strlen... maybe add a note for that? 🤔

Reply:
I removed this line.

Reply:
> If there is \0 in the middle, it will already mess up there with the char* => string conversion and use strlen

Nope, you can do `std::string{"ab\0cde"}` and that'll make the string six characters long with a zero in the middle.